pkg/controller/v1beta1/inferenceservice/controller.go

```diff
@@ -139,15 +142,17 @@ func (r *InferenceServiceReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		// The object is not being deleted, so if it does not have our finalizer,
 		// then lets add the finalizer and update the object. This is equivalent
 		// registering our finalizer.
-		if !utils.Includes(isvc.ObjectMeta.Finalizers, finalizerName) {
-			isvc.ObjectMeta.Finalizers = append(isvc.ObjectMeta.Finalizers, finalizerName)
-			if err := r.Update(context.Background(), isvc); err != nil {
+		if !controllerutil.ContainsFinalizer(isvc, finalizerName) {
+			controllerutil.AddFinalizer(isvc, finalizerName)
+			patchYaml := "metadata:\n  finalizers: [" + strings.Join(isvc.ObjectMeta.Finalizers, ",") + "]"
```

Comment:
Works for no finalizers as well.

Reply:
noob question, is it the best way to create a patch?

Reply:
There are multiple ways to patch, depends on the `types.xxxxPatchType`. This approach with `MergePatchType` will never go wrong - It clearly specifies patching `finalizers` only, and replacing everything under it with the values specified.

Reply:
Sorry, I should be more specific. 
I tried https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client#MergeFrom so I don't need to write the patch yaml on my own.
```
original := client.MergeFrom(localModel.DeepCopy())
localModel.ObjectMeta.Finalizers = utils.RemoveString(localModel.ObjectMeta.Finalizers, finalizerName)
if err := c.Patch(context.TODO(), localModel, original); err != nil {
	return ctrl.Result{}, err
}
```
I got this approach from
https://stackoverflow.com/questions/57712941/what-is-the-proper-way-to-patch-an-object-with-controller-runtime
Your code looks good to me too! Thanks!

Reply:
Thanks, understood.